### ISSUE TITLE
Exited (100) with ENV LANG

### ISSUE THREAD

**Issue by robertogyn19** [dockerfile/mongodb#12]
The container does not start when configure env LANG="en-US.UTF-8".

``` shell
$ docker run -d -e "LANG=en_US.UTF-8" --name mongodb dockerfile/mongodb
$ docker ps -a
CONTAINER ID        IMAGE                                               COMMAND                CREATED             STATUS                        PORTS               NAMES
ea6e30400da3        dockerfile/mongodb:latest                           "mongod"               2 seconds ago       Exited (100) 2 seconds ago                        mongodb
```

``` shell
$ docker version
Client version: 1.2.0
Client API version: 1.14
Go version (client): go1.3.1
Git commit (client): fa7b24f
OS/Arch (client): linux/amd64
Server version: 1.2.0
Server API version: 1.14
Go version (server): go1.3.1
Git commit (server): fa7b24f

$ docker images dockerfile/mongodb
REPOSITORY           TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
dockerfile/mongodb   latest              a0775ca89ae7        2 hours ago         710.7 MB
```

**Comment by jtmarmon** [comment]
@pilwon any idea about this? I'm getting exit code 100 without any env variables (run command is `sudo docker run -d -p 27017:27017 --name=mongo dockerfile/mongodb`)

**Comment by immutef** [comment]
same here, container always exits with code 100 :(

**Comment by jeremy6d** [comment]
same here, the image used to work but now I'm getting exits w/ code 100

**Comment by jaydp17** [comment]
May be it's because of Insufficient free space for journal files. It may not be the case for everyone, but freeing up some space worked for me

**Comment by jtmarmon** [comment]
^ i believe my solution to this was running with `--smallFiles`

**Comment by sunao-uehara** [comment]
Same issue here. --smallFiles doesn't fix this issue for me. Anybody still having issue?

my docker-compose.yml file is like this

``` yaml
database:
  image: mongo:2.6
  command: "--smallfiles"
  volumes:
    - data/mongodb:/data/db
  ports:
    - "27017:27017"
```

**Comment by agamrafaeli** [comment]
For me the problem was that I shut down my computer without shutting down the container properly. 

When starting the container an internal problem happened with the mongo. I SSH'ed into the machine and deleted a lock file and then the container started all right for me.

**Comment by Secbone** [comment]
same here, `docker run` is ok but exits with code 100 with docker-compose :(

**Comment by m-amr** [comment]
try to make the whole data directory as a volume
instead of

volumes:
- ./data/db:/data/db

try
volumes:
- ./data:/data

it works for me.

**Comment by panneerboss** [comment]
work around available - https://github.com/docker-library/mongo/issues/74

**Comment by jovanialferez** [comment]
thanks @m-amr! works for me.

**Comment by m-amr** [comment]
@jovanialferez 
I think it is not working as expected if you try to open data/db you will found it empty.
please check it ?
I have deleted my solution because it is not working as expected.

**Comment by jovanialferez** [comment]
@m-amr yes, it works.. it created `data/mongo` on my host  (windows 10 pro) which maps to `data/db` inside the container which is not empty in my case.

**Comment by jovanialferez** [comment]
@m-amr here is a snippet from my `docker-compose.yml` file:

    mongo:
        image: mongo:latest
        ports:
            - "27017:27017"
    
    mongoclient: 
        image: mongoclient/mongoclient
        ports:
            - "3000:3000"
    volumes_data:
        image: tianon/true
        volumes:
            - ./data:/data

**Comment by m-amr** [comment]
@jovanialferez but the db directory is empty ? can you check it ?

**Comment by jovanialferez** [comment]
@m-amr the db directory inside the container is not empty.

**Comment by jovanialferez** [comment]
@m-amr some files inside the container isnt visible in the host though.

![docker-mongo-mounted-volume](https://user-images.githubusercontent.com/1515234/30240911-f404f14a-95ab-11e7-9012-d011b4844290.PNG)

**Comment by alex1012** [comment]
@m-amr i have same problem, this solution is not working for me =(

**Comment by natesire** [comment]
Is there a way to tell docker or mongodb to exit with a more useful message than just the error code 100. That code could mean a plethora of problems. Unfortunately the docker and mongodb logs are empty.
